### PR TITLE
Feat/ecologist exp

### PR DESCRIPTION
**Pull request by sfaulkner89**

### CODE REVIEW

File: packages/api/src/handlers/ecologist-experience/ecologist-experience-proc.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+export const prepareResponse = a => {
+  Object.assign((({
+    ...experience
+  }) => experience)(a), {
+    createdAt: a.createdAt.toISOString(),
+    updatedAt: a.updatedAt.toISOString()
+  })
+  a.ecologistExperience = a.experience
+  delete a.experience
+  delete a.ecologistExperience.complete
```
Comment (GrahamMWillis) [inline packages/api/src/handlers/ecologist-experience/ecologist-experience-proc.js:15]: There is a new way of handling the completed flags. 
See the application tags:
APIRequests.APPLICATION.tags(applicationId).add('tag')
APIRequests.APPLICATION.tags(applicationId).has('tag') // True
APIRequests.APPLICATION.tags(applicationId).remove('tag') 
APIRequests.APPLICATION.tags(applicationId).has('tag') // False

Comment (GrahamMWillis) [reply packages/api/src/handlers/ecologist-experience/ecologist-experience-proc.js:15]: Then in the tasklist see how the tags are used to drive the status badges

File: packages/database-model/src/sequelize-model.js [hunk 2]
```diff
@@ -96,6 +96,28 @@ async function defineSites (sequelize) {
   })
 }
 
+async function defineEcologistExperience (sequelize) {
+  models.ecologistExperience = await sequelize.define('ecologist-experience', {
+    id: { type: DataTypes.UUID },
+    experience: { type: DataTypes.JSONB },
+    applicationId: {
+      type: DataTypes.UUID,
+      primaryKey: true,
```
Comment (GrahamMWillis) [inline packages/database-model/src/sequelize-model.js:105]: Surely id is the primary key and applicationId is a foreign key?

File: packages/database-model/src/sequelize-model.js [hunk 3]
```diff
@@ -96,6 +96,28 @@ async function defineSites (sequelize) {
...
+      references: {
+        model: models.applications,
+        key: 'id'
+      }
+    },
+    submitted: { type: DataTypes.DATE },
```
Comment (GrahamMWillis) [inline packages/database-model/src/sequelize-model.js:111]: Probably not necessary as the experience will be submitted at the same time as the application, but leave it for now.

File: packages/web-service/src/pages/ecologist-experience/enter-experience/enter-experience.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import Joi from 'joi'
+import pageRoute from '../../../routes/page-route.js'
+import { APIRequests } from '../../../services/api-requests.js'
+import { ecologistExperienceURIs } from '../../../uris.js'
+
+export const completion = async request => {
+  const journeyData = await request.cache().getData()
+  if (journeyData.ecologistExperience.complete) {
```
Comment (GrahamMWillis) [inline packages/web-service/src/pages/ecologist-experience/enter-experience/enter-experience.js:8]: See comment on tags above

File: packages/web-service/src/pages/ecologist-experience/enter-experience/enter-experience.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  const journeyData = await request.cache().getData()
+  return journeyData.ecologistExperience.experienceDetails
+}
+
+export const setData = async request => {
+  const pageData = await request.cache().getPageData()
+  const journeyData = await request.cache().getData()
+  const experienceDetails = pageData.payload[ecologistExperienceURIs.ENTER_EXPERIENCE.page]
+  Object.assign(journeyData.ecologistExperience, { experienceDetails })
+  if (journeyData.ecologistExperience.complete) {
```
Comment (GrahamMWillis) [inline packages/web-service/src/pages/ecologist-experience/enter-experience/enter-experience.js:24]: See comments on tags above

File: packages/web-service/src/pages/ecologist-experience/enter-experience/enter-experience.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    APIRequests.ECOLOGIST_EXPERIENCE.putExperienceById(journeyData.applicationId, journeyData.ecologistExperience)
+  }
+  await request.cache().setData(journeyData)
+}
+
+export default pageRoute({
+  uri: ecologistExperienceURIs.ENTER_EXPERIENCE.uri,
+  page: ecologistExperienceURIs.ENTER_EXPERIENCE.page,
+  validator: Joi.object({
+    'enter-experience': Joi.string()
```
Comment (GrahamMWillis) [inline packages/web-service/src/pages/ecologist-experience/enter-experience/enter-experience.js:34]: Joi objects are optional by default I think so you may want .required() // check the docs though

File: packages/web-service/src/pages/ecologist-experience/enter-license-details/enter-license-details.njk [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    name: "enter-license-details",
+    hint: {
+      text: "Enter the reference number of any previously held badger mitigation license."
+    },
+    errorMessage: { text: "Enter the reference number of your previously held licence" } if error['enter-license-details'],
+    classes: "govuk-input--width-10",
+    spellcheck: false
+  }) }}
+
+{% endblock %}
```
Comment (GrahamMWillis) [inline packages/web-service/src/pages/ecologist-experience/enter-license-details/enter-license-details.njk:27]: Watch these end of lines. Not a big deal but the code quality tools like a CR/New line to terminate a file

File: packages/web-service/src/pages/tasklist/licence-type-map.js [hunk 8]
```diff
@@ -18,6 +17,7 @@ export const SECTION_TASKS = {
   PERMISSIONS: 'permissions',
   SITES: 'sites',
   SETTS: 'setts',
+  ECOLOGIST_EXPERIENCE: 'ecologist-experience',
```
Comment (GrahamMWillis) [inline packages/web-service/src/pages/tasklist/licence-type-map.js:20]: See how the status tags are used in eligibility

**Review by GrahamMWillis — CHANGES_REQUESTED** [review]

**Review by kealanparr — APPROVED** [review]

File: packages/api/src/handlers/ecologist-experience/ecologist-experience-proc.js (hunk 1 above)
Comment (sfaulkner89) [reply packages/api/src/handlers/ecologist-experience/ecologist-experience-proc.js:15]: All sorted on the tags

**Review by GrahamMWillis — APPROVED** [review]

**Pull request merged**